content/applications/general/product_images.rst

```diff
@@ -0,0 +1,133 @@
+------------------------------------
+
+#. Go to `Google Programmable Search Engine <https://programmablesearchengine.google.com/>`__ and
+   click on `Get Started`. Log in with your Google account.
+
+   .. image:: product_images/google-images-credentials03.png
+      :align: center
+
+#. Fill the language and the name of the search engine. Give it an explicit name
+   (e.g. Odoo Images).
+
+   .. note::
+      Google doesn't allow to create a search engine without having entered at least one specific
+      site to search on. You can put any website (e.g. www.google.com) for this step, we will
+      remove it later.
+
+#. Validate the form by clicking on `Create`. Then, go to the edition mode of the search engine
+   that you created (either by clicking on `Control Panel` on the confirmation page or by
+   clicking on the name of your Search Engine on the Home page).
+
+#. In the basics tab, make sure to enable Image search, SafeSearch and Search the entire web.
+
+   .. note::
+      Once `Search the entire web` enabled, you can safely delete the site that you previously
+      entered at the previous step.
+
+#. Save your **Search Engine Id**.
+
+   .. tip::
+      You can easily save your Search Engine ID by clicking on the Copy to clipboard button next to
+      it.
+
+.. _product_images/setup-in-odoo:
+
+Odoo
+----
+
+#. Go to :menuselection:`Settings --> General Settings --> Integrations`,
+   activate **Google Images** and save.
+
+#. Go back to :menuselection:`Settings --> General Settings`, enter your **API Key** and
+   **Search Engine ID** in Google Images settings and save again.
+
+#. The setup is now ready.
+
+.. _product_images/get-product-images:
+
+Automatically get your product images in Odoo
+=============================================
+
+The action to automatically get your product images in Odoo appears in any Products or Product
+Variants list view. Here is a step-by-step guide from the Inventory app.
+
+#. Go to the Products menu (:menuselection:`Products --> Products` or :menuselection:`Products --> 
+   Product Variants`) from any application that uses products like Inventory or Sale.
+
+#. On the list view, select the products that needs an image.
+
+   .. important::
+      Only the 10K first products or product variants selected will be processed.
+
+   .. note::
+      Only the products or product variants with a barcode and without an image will be processed.
+      If you select a product that has one or more variants from the Products view, each variant
+      matching the previous criteria will be processed.
+
+#. In the action menu, select `Get Pictures from Google Images` and validate by clicking on 
+   `Get picture`.
+
+#. You should see your images appearing incrementally.
+
+   .. note::
+      Only the 10 first images are fetched immediatly. If you selected more than 10, the rest will
+      be fetched as a background job, so you can continue doing your work while illustrating your
+      products.
```

Comment:
```suggestion
      be fetched as a background job.
```
It sounds like you can resume painting your products while the cron fetches the images.

Reply:
![Painting my forehead](https://64.media.tumblr.com/87862fca1f104ca2c42f8784689930ec/tumblr_n0idxh3bL21s01qkyo1_500.gifv)